src/main/scala/sbtavro/SbtAvro.scala

```diff
@@ -138,18 +199,22 @@ object SbtAvro extends AutoPlugin {
 
   private def sourceGeneratorTask(key: TaskKey[Seq[File]]) = Def.task {
     val out = (key / streams).value
+    val externalSrcDir = (avroUnpackDependencies / target).value
     val srcDir = (key / sourceDirectory).value
     val outDir = (key / sourceManaged).value
     val strType = avroStringType.value
     val fieldVis = avroFieldVisibility.value
     val enbDecimal = avroEnableDecimalLogicalType.value
     val useNs = avroUseNamespace.value
-    val cachedCompile = FileFunction.cached(out.cacheDirectory / "avro",
-      inStyle = FilesInfo.lastModified,
-      outStyle = FilesInfo.exists) { (in: Set[File]) =>
-        compileAvroSchema(srcDir, outDir, out.log, strType, fieldVis, enbDecimal, useNs)
-      }
-    cachedCompile((srcDir ** AvroFilter).get.toSet).toSeq
+
+    def cachedCompile(dir: File) = {
+      val schemas = (dir ** AvroFilter).get.toSet
+      FileFunction.cached(out.cacheDirectory / "avro", FilesInfo.lastModified, FilesInfo.exists) { _ =>
+        compileAvroSchema(dir, outDir, out.log, strType, fieldVis, enbDecimal, useNs)
+      }(schemas)
+    }
+
+    Seq(externalSrcDir, srcDir).flatMap(cachedCompile)
```

Comment:
This may break caching. I need to check how `FileFunction.cached` works internally